k8sgpt v0.3.11 or later crashes on Linux when running --filter ValidatingWebhookConfiguration

### Checklist

- [X] I've searched for similar issues and couldn't find anything matching
- [X] I've included steps to reproduce the behavior

### Affected Components

- [X] K8sGPT (CLI)
- [ ] K8sGPT Operator

### K8sGPT Version

v0.3.14

### Kubernetes Version

v1.25.11+1485cc9

### Host OS and its Version

Ubuntu 22.04.3, RHEL 9.2

### Steps to reproduce

1. Install (or update to) k8sgpt v0.3.14 on Linux (tested on Ubuntu and RHEL 9.2)
2. Run an analyze command

Red Hat Enterprise Linux 9:
```
$ k8sgpt analyze -e
panic: runtime error: slice bounds out of range [:-1]

goroutine 33 [running]:
github.com/k8sgpt-ai/k8sgpt/pkg/util.MapToString(0x0?)
        /home/runner/work/k8sgpt/k8sgpt/pkg/util/util.go:220 +0x165
github.com/k8sgpt-ai/k8sgpt/pkg/analyzer.ValidatingWebhookAnalyzer.Analyze({}, {0xc000a64480, {0x2c6a2c0, 0xc000120028}, {0x0, 0x0}, {0x2c6a9f8, 0x3fb9840}, 0x0, {0x0, ...}, ...})
        /home/runner/work/k8sgpt/k8sgpt/pkg/analyzer/validating_webhook.go:81 +0x988
github.com/k8sgpt-ai/k8sgpt/pkg/analysis.(*Analysis).RunAnalysis.func1({0x2c4e520, 0x3ff8de0}, 0x0?, 0x0?)
        /home/runner/work/k8sgpt/k8sgpt/pkg/analysis/analysis.go:171 +0xd8
created by github.com/k8sgpt-ai/k8sgpt/pkg/analysis.(*Analysis).RunAnalysis
        /home/runner/work/k8sgpt/k8sgpt/pkg/analysis/analysis.go:169 +0xa1d
```
Ubuntu 22.04:
```
$ k8sgpt analyze -e
panic: runtime error: slice bounds out of range [:-1]

goroutine 32 [running]:
github.com/k8sgpt-ai/k8sgpt/pkg/util.MapToString(0x0?)
        /home/runner/work/k8sgpt/k8sgpt/pkg/util/util.go:220 +0x165
github.com/k8sgpt-ai/k8sgpt/pkg/analyzer.ValidatingWebhookAnalyzer.Analyze({}, {0xc0004a0c30, {0x2c6a2c0, 0xc00011a028}, {0x0, 0x0}, {0x2c6a9f8, 0x3fb9840}, 0x0, {0x0, ...}, ...})
        /home/runner/work/k8sgpt/k8sgpt/pkg/analyzer/validating_webhook.go:81 +0x988
github.com/k8sgpt-ai/k8sgpt/pkg/analysis.(*Analysis).RunAnalysis.func1({0x2c4e520, 0x3ff8de0}, 0x0?, 0x0?)
        /home/runner/work/k8sgpt/k8sgpt/pkg/analysis/analysis.go:171 +0xd8
created by github.com/k8sgpt-ai/k8sgpt/pkg/analysis.(*Analysis).RunAnalysis
        /home/runner/work/k8sgpt/k8sgpt/pkg/analysis/analysis.go:169 +0xa1d
```

### Expected behaviour

Getting any working output

### Actual behaviour

a panic runtime error

### Additional Information

_No response_

I've ran all filters one by one, and only 1 of them produced this error.
```
pk@host01:~$ ./k8sgpt version
k8sgpt: 0.3.11 (ee83aa1), built at: unknown
pk@host01:~$ ./k8sgpt analyze --filter MutatingWebhookConfiguration
AI Provider: openai

No problems detected

pk@host01:~$ ./k8sgpt analyze --filter ValidatingWebhookConfiguration
AI Provider: openai

No problems detected

panic: runtime error: slice bounds out of range [:-1]

goroutine 13 [running]:
github.com/k8sgpt-ai/k8sgpt/pkg/util.MapToString(0x0?)
        /home/runner/work/k8sgpt/k8sgpt/pkg/util/util.go:220 +0x165
github.com/k8sgpt-ai/k8sgpt/pkg/analyzer.ValidatingWebhookAnalyzer.Analyze({}, {0xc000a2f500, {0x2a780d0, 0xc000054128}, {0x0, 0x0}, {0x2a787d0, 0x3d99c60}, 0x0, {0x0, ...}, ...})
        /home/runner/work/k8sgpt/k8sgpt/pkg/analyzer/validating_webhook.go:79 +0x9a5
github.com/k8sgpt-ai/k8sgpt/pkg/analysis.(*Analysis).RunAnalysis.func2({0x2a5cdc0?, 0x3dd91b8?}, {0xc0006474e0, 0x1e})
        /home/runner/work/k8sgpt/k8sgpt/pkg/analysis/analysis.go:198 +0xe2
created by github.com/k8sgpt-ai/k8sgpt/pkg/analysis.(*Analysis).RunAnalysis
        /home/runner/work/k8sgpt/k8sgpt/pkg/analysis/analysis.go:196 +0x8df

```

--- a/pkg/analyzer/mutating_webhook.go
+++ b/pkg/analyzer/mutating_webhook.go
@@ -78,6 +78,10 @@ func (MutatingWebhookAnalyzer) Analyze(a common.Analyzer) ([]common.Result, erro
 				continue
 			}
 
+			// When Service selectors are empty we defer to service analyser
+			if len(service.Spec.Selector) == 0 {
+				continue
+			}
 			// Get pods within service
 			pods, err := a.Client.GetClient().CoreV1().Pods(svc.Namespace).List(context.Background(), v1.ListOptions{
 				LabelSelector: util.MapToString(service.Spec.Selector),

--- a/pkg/analyzer/validating_webhook.go
+++ b/pkg/analyzer/validating_webhook.go
@@ -76,6 +76,10 @@ func (ValidatingWebhookAnalyzer) Analyze(a common.Analyzer) ([]common.Result, er
 				continue
 			}
 
+			// When Service selectors are empty we defer to service analyser
+			if len(service.Spec.Selector) == 0 {
+				continue
+			}
 			// Get pods within service
 			pods, err := a.Client.GetClient().CoreV1().Pods(svc.Namespace).List(context.Background(), v1.ListOptions{
 				LabelSelector: util.MapToString(service.Spec.Selector),

